src/main/scala/outwatch/dom/helpers/DomUtils.scala

```diff
@@ -123,65 +124,64 @@ object DomUtils {
       .pairwise
       .subscribe(tuple => patch(tuple._1, tuple._2), console.error(_))
 
-    subscriptionPromise.success(subscription)
+    subscriptionRef.put(subscription).unsafeRunSync()
 
     proxy.elm.foreach((e: Element) => hooks.foreach(_.sink.next(e)))
   }
 
-  private def createDestroyHook(subscription: Future[Subscription], hooks: Seq[DestroyHook]) = (proxy: VNodeProxy) => {
-    import scala.concurrent.ExecutionContext.Implicits.global
-
+  private def createDestroyHook(subscription: STRef[Subscription], hooks: Seq[DestroyHook]) = (proxy: VNodeProxy) => {
     proxy.elm.foreach((e: Element) => hooks.foreach(_.sink.next(e)))
-    subscription.foreach(_.unsubscribe())
+    subscription.update { s => s.unsubscribe(); s }.unsafeRunSync()
+    ()
   }
 
 
   private[outwatch] final case class SeparatedModifiers(
-    emitters: Seq[Emitter] = Seq.empty[Emitter],
-    receivers: Seq[Receiver] = Seq.empty[Receiver],
-    properties: Seq[Property] = Seq.empty[Property],
-    vNodes: Seq[VNode] = Seq.empty[VNode]
+    emitters: List[Emitter] = List.empty[Emitter],
```

Comment:
Maybe just write `Nil` here? It is `Nil` anyways:https://github.com/scala/scala/blob/v2.12.3/src/library/scala/collection/immutable/List.scala#L463

And maybe move this change into a separate commit.